zerver/webhooks/githubsponsors/view.py

```diff
@@ -0,0 +1,174 @@
+from datetime import datetime, timezone
+from typing import Callable, Dict, Optional
+
+from django.http import HttpRequest, HttpResponse
+
+from zerver.decorator import webhook_view
+from zerver.lib.exceptions import UnsupportedWebhookEventTypeError
+from zerver.lib.response import json_success
+from zerver.lib.typed_endpoint import JsonBodyPayload, typed_endpoint
+from zerver.lib.validator import WildValue, check_string
+from zerver.lib.webhooks.common import (
+    OptionalUserSpecifiedTopicStr,
+    check_send_webhook_message,
+    get_http_headers_from_filename,
+    validate_extract_webhook_http_header,
+)
+from zerver.models import UserProfile
+
+fixture_to_headers = get_http_headers_from_filename("HTTP_X_GITHUB_EVENT")
+
+
+class Helper:
+    def __init__(
+        self,
+        request: HttpRequest,
+        payload: WildValue,
+        include_title: bool,
+    ) -> None:
+        self.request = request
+        self.payload = payload
+        self.include_title = include_title
+
+
+def get_cancelled_body(helper: Helper) -> str:
+    payload = helper.payload
+    template = "{user_name} cancelled their {subscription} subscription."
+    return template.format(
+        user_name=get_sender_name(payload),
+        subscription=get_subscription(payload),
+    ).rstrip()
```

Comment:
Probably you can just drop the usage of the `Helper` class here as its properties such as `include_title` are not being used anywhere.

```suggestion
def get_cancelled_body(payload: WildValue) -> str:
    template = "{user_name} cancelled their {subscription} subscription."
    return template.format(
        user_name=get_sender_name(payload),
        subscription=get_subscription(payload),
    ).rstrip()
```

Reply:
Yeah, that's good point. I was using the GitHub integration as template, and thought these properties might come in handy to add more functionality to this integration in the future. Is it bad practice to leave them there for future potential use cases?

Reply:
We can leave it for future potential but I don't think it would take huge refactoring to implement something like this. Right now, it's just taking one more mental step to understand the whole flow and hence isn't useful here.